testsuite/pytests/test_sp/test_sp_manager.py

```diff
@@ -59,7 +59,8 @@ def test_register_synapses(self):
                 nest.structural_plasticity_synapses = {"syn1": syn_dict}
                 kernel_status = nest.structural_plasticity_synapses
                 assert "syn1" in kernel_status
-                assert kernel_status["syn1"] == extract_dict_a_from_b(kernel_status["syn1"], syn_dict)
+                for kv in extract_dict_a_from_b(kernel_status["syn1"], syn_dict).items():
+                    assert kv in kernel_status["syn1"].items()
```

Comment:
I don't understand what is going on here.

Reply:
This just checks that all key/value pairs from the outer dict (the one that was set) are contained in the inner dict (the one that was retrieved after setting).  This broke due to the addition of new values that are present in the retrieved dict with their defaults even though they might not have been present in the set dict.

In general: A lot of the testing for the structural plasticity code is rather simple and not using modern techniques and someone should clean this up real good. I just don't think I can do this right now.